lancero_source.go

```diff
@@ -569,7 +577,18 @@ func (ls *LanceroSource) getNextBlock() chan *dataBlock {
 				panic(fmt.Sprintf("timeout, no data from lancero after %v / %v", panicTime, ls.readPeriod))
 
 			case mfo := <-ls.mixRequests:
-				ls.Mix[mfo.ChannelIndex].errorScale = mfo.MixFraction / float64(ls.nsamp)
+				for i, index := range mfo.ChannelIndices {
+					if index < 0 || index >= len(ls.Mix) {
+						continue
+					}
```

Comment:
Don't silently ignore incorrect input, throw an error, or panic. Also it was checked in `ConfigureMixFraction`. I would just removed this check and let it panic on out of bounds.